### PR TITLE
Python: Refactor to use common decimal and datetime util

### PR DESCRIPTION
**Pull request by rdblue**
This fixes a couple of minor comments from #4416 and refactors to use common decimal and datetime util functions.

### CODE REVIEW

**Comment by jun-he** [comment]
SGTM, thanks @rdblue

File: python/src/iceberg/utils/decimal.py [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+    if isinstance(value, int):
+        return (value.bit_length() + 7) // 8
+    elif isinstance(value, Decimal):
+        return (decimal_to_unscaled(value).bit_length() + 7) // 8
+
+    raise ValueError(f"Unsupported value: {value}")
+
+
+def decimal_to_bytes(value: Decimal) -> bytes:
+    unscaled_value = decimal_to_unscaled(value)
```
Comment (samredai) [inline python/src/iceberg/utils/decimal.py:69]: Suggesting a docstring here:

    """Returns a byte representation of a decimal

    Args:
        value (Decimal): a decimal value
    Returns:
        bytes: A byte representation of the decimal unscaled
    """

**Review by samredai — APPROVED** [review]
One suggestion but otherwise LGTM

File: python/src/iceberg/utils/decimal.py (hunk 1 above)
Comment (rdblue) [reply python/src/iceberg/utils/decimal.py:76]: Thanks! I updated this.

File: python/src/iceberg/utils/decimal.py [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+        scale (int): A scale to set for the returned Decimal instance
+
+    Returns:
+        Decimal: A scaled Decimal instance
+    """
+    sign, digits, _ = Decimal(unscaled).as_tuple()
+    return Decimal((sign, digits, -scale))
+
+
+def min_size(value: Union[int, Decimal]) -> int:
```
Comment (danielcweeks) [inline python/src/iceberg/utils/decimal.py:51]: Should we call this `min_bytes`?

Comment (rdblue) [reply python/src/iceberg/utils/decimal.py:51]: To me, that implies that this returns some min as bytes, rather than a size. What about `required_size`?

Comment (rdblue) [reply python/src/iceberg/utils/decimal.py]: Updated to `bytes_required`.

**Comment by rdblue** [comment]
Thanks for reviewing, @jun-he, @samredai, and @danielcweeks!

**Pull request merged**